src/gallium/drivers/lima/lima_resource.c

```diff
@@ -222,17 +249,24 @@ lima_resource_get_handle(struct pipe_screen *pscreen,
                          struct pipe_resource *pres,
                          struct winsys_handle *handle, unsigned usage)
 {
+   struct lima_screen *screen = lima_screen(pscreen);
    struct lima_resource *res = lima_resource(pres);
    lima_bo_handle bo = res->buffer->bo;
    int err;
 
    switch (handle->type) {
    case DRM_API_HANDLE_TYPE_SHARED:
+      if (screen->ro) {
+         fprintf(stderr, "flink unsupported with sun4i-drm\n");
+         return FALSE;
+      }
       err = lima_bo_export(bo, lima_bo_handle_type_gem_flink_name, &handle->handle);
       if (err)
          return FALSE;
       break;
    case DRM_API_HANDLE_TYPE_KMS:
+      if (screen->ro && renderonly_get_handle(res->scanout, handle))
```

Comment:
Can you confirm do we really need this? Because I think lima_bo_export is enough here.

Reply:
We do need this, at some point renderonly calls resource_get_handle with DRM_API_HANDLE_TYPE_KMS.